nautobot/core/testing/integration.py

```diff
@@ -220,3 +253,328 @@ def fill_select2_multiselect_field(self, field_name, value):
         # wait for "searching" to disappear
         self.browser.is_element_not_present_by_css(".loading-results", wait_time=5)
         search_box.first.type(Keys.ENTER)
+
+    def click_button(self, query_selector):
+        btn = self.browser.find_by_css(query_selector, wait_time=5)
+        # Button might be visible but on the edge and then impossible to click due to vertical/horizontal scrolls
+        self.browser.execute_script(f"document.querySelector('{query_selector}').scrollIntoView()")
+        btn.click()
+
+    def login_as_superuser(self):
+        self.user.is_superuser = True
+        self.user.save()
+        self.login(self.user.username, self.password)
+
+
+class BulkOperationsTestCases:
+    """
+    Helper classes that runs all the basic bulk-operations test cases like edit/delete with
+    filtered / not filtered items along with select all option.
+
+    To use this class create required items in setUp method:
+    - at least four entities in two different groups,
+    - provide field for filtering to distinguish between above two groups
+    - provide expected counts (if different from default)
+    - set edit field and value
+    """
+
+    class BaseTestCase(SeleniumTestCase):
+        model_menu_path = (None, None)
+        model_base_viewname = ""
+        model_edit_data = {}
+        model_filter_by = {}
+        model_class = None
+        model_plural = None
```

Comment:
If these are mandatory for a subclass to define, and don't work with the given defaults, let's use typing to declare them as such with no defaults, perhaps:

```suggestion
        model_menu_path: tuple[str, str]
        model_base_viewname: str
        model_edit_data: dict[str, Any]
        model_filter_by: dict[str, Any]
        model_class: type[Model]
        model_plural: str
```